### ISSUE TITLE
Start-SqlMigration login errors with Credentials

### ISSUE THREAD

**Issue by elguapo79** [sqlcollaborative/dbatools#1190]
Trying to migrate across domains, so I'm using a SQL user, with -SourceSqlCredential, and I can tell it validates because if I intentionally put the wrong PW in it tells me it failed, but when it tries to actually do anything, all it says is Login failed for user ''. The user is not associated with a trusted SQL Server Connection.  I double checked by running Connect-DbaSqlServer and that works fine also.

Attaching debug output as requested by Fred
[output.zip](https://github.com/sqlcollaborative/dbatools/files/993850/output.zip)

**Comment by FriedrichWeinmann** [comment]
Hi,
sorry for getting back at you only after 3 days.
I checked the data you sent and can totally see the same things you described in the logs.
Much as I hate to admit it, it didn't make much sense to me from looking at the source - should have worked without a hitch.

When you connect using `Connect-DbaSqlServer`, do you use the default settings (that is: IP Address and Credentials) and let the function do the rest with default settings?

If that's it, I'll have to ask someone deeper on the DBA end of business look at the fine-print in the differences between the public function and what we use internally (not quite the same thing).

That said, you can use the connection object `Connect-DbaSqlServer` returns as input to `Start-SqlMigration`. In that case simply don't specify credentials, our commands will reuse that connection object to access the source server.

**Comment by elguapo79** [comment]
Yes, that’s correct – using Connect-DbaSqlServer I just used the IP and credentials and it worked fine.

I tried as you mentioned below, still getting the same error.  Check out the screenshot.  It connects, but gives the error on the Migration, but then I can still use the object I created and query off it, so the connection is working, but for some reason the migration script doesn’t like it

[cid:image002.png@01D2CD60.E1F01D90]

Jason Allen

**Comment by Zhunya** [comment]
Have same cross-domain authentication problems when trying to Copy-DbaDatabase/Restore-DbaDatabase. 

From my machine:
![windows powershell 2017-06-15 17 35 00](https://user-images.githubusercontent.com/14327077/27186812-1ecf5684-51f2-11e7-91ca-765952dade7c.png)

On remote host:
![snipping tool 2017-06-15 17 42 17](https://user-images.githubusercontent.com/14327077/27186842-345d1810-51f2-11e7-95ba-1cbfbafa45dc.png)

**Comment by wsmelton** [comment]
I have an PR submitted where this is potentially going to resolve this issue. Once @potatoqualitee (or other with rights) merges that PR linked please give this another try since we do not have easy access to a cross-domain environment in our test lab(s).

**Comment by potatoqualitee** [comment]
merged-a-rama in dev

**Comment by niphlod** [comment]
fixed via https://github.com/sqlcollaborative/dbatools/pull/2023

**Comment by getstoopid** [comment]
Sorry, not fixed as of version 0.9.811 - Connect-DbaInstance still throws "login failed for user '[current windows user]' despite providing sqlcredentials.

**Comment by potatoqualitee** [comment]
need way more info, please open a new issue @getstoopid
